src/client/generic/components/form/input/Select.tsx

```diff
@@ -23,8 +23,8 @@ export function Select<T>({
       {...props}
       value={selectedValue as any}
     >
-      {options.map(({ label }, idx) => (
-        <option key={idx} value={idx}>
+      {options.map(({ label, value }, idx) => (
+        <option key={idx} value={String(value)}>
```

Comment:
The `String` call is unnecessary 
```suggestion
        <option key={idx} value={value}>
```

Also, if you want to convert a value into string the preferred way is to call the `toString` method